lib/mpl_toolkits/tests/test_mplot3d.py

```diff
@@ -513,3 +513,20 @@ def test_autoscale():
     ax.set_autoscalez_on(True)
     ax.plot([0, 2], [0, 2], [0, 2])
     assert ax.get_w_lims() == (0, 1, -.1, 1.1, -.4, 2.4)
+
+
+@pytest.mark.parametrize('value', [np.inf, np.nan])
+@pytest.mark.parametrize(('setter', 'side'), [
+    ('set_xlim3d', 'left'),
+    ('set_xlim3d', 'right'),
+    ('set_ylim3d', 'bottom'),
+    ('set_ylim3d', 'top'),
+    ('set_zlim3d', 'bottom'),
+    ('set_zlim3d', 'top'),
+])
+def test_invalid_axes_limits(setter, side, value):
+    limit = {side: value}
+    with pytest.raises(ValueError):
+        fig = plt.figure()
+        obj = fig.add_subplot(111, projection='3d')
+        getattr(obj, setter)(**limit)
```

Comment:
Only this line should be in the context, just in case the first two raise other things accidentally.